src/components/App/App.jsx

```diff
@@ -0,0 +1,156 @@
+import { useState, useEffect } from "react";
+import {
+  HashRouter as Router,
+  Routes,
+  Route,
+  useNavigate,
+} from "react-router-dom";
+import "./App.css";
+import Header from "../Header/Header";
+import Profile from "../Profile/Profile";
+import Main from "../Main/Main";
+import Footer from "../Footer/Footer";
+import Bookmark from "../Bookmark/Bookmark";
+import { AuthContext } from "../../contexts/spotifyContext";
+import { refreshAccessToken, getToken } from "../../utils/SpotifyApi";
+
+function App() {
+  const [accessToken, setAccessToken] = useState(null);
+
+  const [loading, setLoading] = useState(true);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [userData, setUserData] = useState(null);
+  const navigate = useNavigate();
+
+  // Generate a random string for the code_verifier
+  const generateRandomString = (length) => {
+    const possible =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    const values = crypto.getRandomValues(new Uint8Array(length));
+    return values.reduce((acc, x) => acc + possible[x % possible.length], "");
+  };
+
+  // Generates the code challenge from the code verifier
+  const generateCodeChallenge = async (codeVerifier) => {
+    const encoder = new TextEncoder();
+    const data = encoder.encode(codeVerifier);
+    const hashBuffer = await window.crypto.subtle.digest("SHA-256", data);
+    const hashArray = new Uint8Array(hashBuffer);
+    return base64UrlEncode(hashArray);
+  };
+
+  // Base64 URL Encoding function
+  const base64UrlEncode = (array) => {
+    return btoa(String.fromCharCode.apply(null, array))
+      .replace(/\+/g, "-")
+      .replace(/\//g, "_")
+      .replace(/=+$/, "");
+  };
+
+  const handleSpotifyLogin = async () => {
+    const codeVerifier = generateRandomString(64);
+    const codeChallenge = await generateCodeChallenge(codeVerifier);
+
+    localStorage.setItem("code_verifier", codeVerifier);
+
+    const client_id = import.meta.env.VITE_SPOTIFY_CLIENT_ID;
+
+    /* const redirectUri =
+      import.meta.env.MODE === "production"
+        ? "https://lupesanchez24.github.io/playlist-frontend/callback" // Production Redirect URI
+        : "http://localhost:3000/callback"; // Localhost Redirect URI */
+
+    const redirectUri =
+      import.meta.env.MODE === "production"
+        ? "https://lupesanchez24.github.io/playlist-frontend/#/callback"
```

Comment:
`https://lupesanchez24.github.io/playlist-frontend` is duplicated 4 times, so you need to create `APP_URL` for that